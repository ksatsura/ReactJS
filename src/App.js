//imports
import React from 'react';
import Header from './Header/Header.js';
import FilmCollection from './FilmCollection/FilmCollection';
import Footer from './Footer/Footer.js';
import ResultsArea from './ResultsArea/ResultsArea';
import ErrorBoundary from './ErrorBoundary/ErrorBoundary';
import { connect } from 'react-redux';
import './style.css';

const App = (props) => {

    const { films, isfilmClicked, genre, filmsSameGenre } = props;
    const filmsNumber = props.films.length;

    return (
        <div className='app'>
            <ErrorBoundary>
                <Header isfilmClicked={isfilmClicked} />
            </ErrorBoundary>
            <ErrorBoundary>
                <ResultsArea isfilmClicked={isfilmClicked} genre={genre} filmsNumber={filmsNumber} />
            </ErrorBoundary>
            <ErrorBoundary>
                <FilmCollection filmsNumber={filmsNumber} films={films} filmsSameGenre={filmsSameGenre} isfilmClicked={isfilmClicked} />
            </ErrorBoundary>
            <Footer/>
        </div>
    );
};

const mapStateToProps = state => {

    return {
        films: state.filmListReducer.films,
        isfilmClicked: state.filmReducer.isfilmClicked,
        genre: state.filmReducer.film.genres[0],
        filmsSameGenre: state.filmListReducer.filmsSameGenre,
    };
};

export default connect(mapStateToProps)(App);