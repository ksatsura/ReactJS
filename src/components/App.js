//imports
import * as React from 'react';
import FilmPage from '../pages/FilmPage.js';
import NoResultsPage from '../pages/NoResultsPage';
import NotFoundPage from '../pages/NotFoundPage';
import SearchPage from '../pages/SearchPage';
import ErrorBoundary from './ErrorBoundary/ErrorBoundary';
import { Route, Switch } from 'react-router-dom';
import styled from 'styled-components';
import { connect } from 'react-redux';
import '../style.css';

const AppDiv = styled.div`
    font-family: Arial;
    width: 100%;
    height: 100%;
`;

// @flow
type Props = {
    films: Array<{}>, 
    genre: string, 
    filmsSameGenre: Array<{}>,
    match: {
        params: {
            id: string
        }
    },
    handleUrlChange: (string) => void
};

class App extends React.Component<Props> {

    componentDidMount() {
        const id = this.props.match.params.id;
        if (id) {
            this.props.handleUrlChange(this.props.genre, id);
        }
    }

    render() {

        return (
            <AppDiv>
                <ErrorBoundary>
                    <Switch>
                        <Route exact path='/' component={NoResultsPage} />
                        <Route path='/film/:id' component={FilmPage} />
                        <Route path='/search/:value' component={SearchPage} />
                        <Route path='/' component={NotFoundPage} />
                    </Switch>
                </ErrorBoundary>
            </AppDiv>  
        );
    }
}

const mapStateToProps = state => {

    return {
        genre: (state.filmReducer.film.genres && state.filmReducer.film.genres[0]) || 'Drama',
    };
};

const mapDispatchToProps = dispatch => {
    return {
        handleUrlChange: (genre, id) => { dispatch(fetchRequestIfNeeded(genre, true, id)); },
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);





