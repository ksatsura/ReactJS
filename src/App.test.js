import React from 'react';
import App from './App.js';
import { shallow } from 'enzyme';

describe('App.js', () => {

    const props = {
        assets: [{}, {}], 
        isAssetClicked: false, 
        genre: ' ', 
        assetsSameGenre: [{}, {}]
    };

    describe('renders', () => {

        it('renders for desktop', () => {
            const component = shallow(<App {...props} />);

            expect(component).toMatchSnapshot();
        });

        it('renders for mobile', () => {
            const component = shallow(<App {...props} isMobile />);

            expect(component).toMatchSnapshot();
        });
    });

});
