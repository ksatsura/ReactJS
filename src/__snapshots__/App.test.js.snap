// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js should render as expected 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
